offload/plugins-nextgen/common/src/PluginInterface.cpp

```diff
@@ -1533,6 +1533,55 @@ Error GenericDeviceTy::printInfo() {
   return Plugin::success();
 }
 
+Expected<GenericKernelTy &>
+GenericDeviceTy::getKernel(llvm::StringRef Name, DeviceImageTy *ImagePtr) {
+
+  GenericKernelTy *&KernelPtr = KernelMap[Name];
+  if (!KernelPtr) {
```

Comment:
I think this is similar handling to the `ctor / dtor` kernels. Maybe we could unify that?

Reply:
Will do.